Automattic/vip-go-mu-plugins
# Two Factor: Better check for proxied Automattician

## Description

Instead of just checking for the value of the `A8C_PROXIED_REQUEST` constant, we can also add the weaker check that the apparent VIP Support User does appear to be an Automattician. In combination with #1615, this will also check that the constant is also defined.

If the weaker check is not desired, then switching the comparison to `is_proxied_request()` would save checking against the constant directly.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Check out PR.
1. Turn off proxy and ensure that Two Factor is requested when logging in as an administrator on a site that requires 2FA for administrators (default).
2. Turn on proxy again, but change VIP Support User to either have a non-Automattic email address, or delete the user meta that marks it as verified. See [here](https://github.com/Automattic/vip-support/blob/f613ab745501a3469b266e5447c694f5d2260931/class-vip-support-user.php#L740-L753).
